File: SDKs/amazonaws.com/ec2/2016-11-15/go/pkg/models/operations/postterminateinstances.go
package operations




type PostTerminateInstancesActionEnum string

const (
    PostTerminateInstancesActionEnumTerminateInstances PostTerminateInstancesActionEnum = "TerminateInstances"
)



type PostTerminateInstancesVersionEnum string

const (
    PostTerminateInstancesVersionEnumTwoThousandAndSixteen1115 PostTerminateInstancesVersionEnum = "2016-11-15"
)


type PostTerminateInstancesQueryParams struct {
    Action PostTerminateInstancesActionEnum `queryParam:"style=form,explode=true,name=Action"`
    Version PostTerminateInstancesVersionEnum `queryParam:"style=form,explode=true,name=Version"`
    
}

type PostTerminateInstancesHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    
}

type PostTerminateInstancesRequest struct {
    QueryParams PostTerminateInstancesQueryParams 
    Headers PostTerminateInstancesHeaders 
    Request []byte `request:"mediaType=text/xml"`
    
}

type PostTerminateInstancesResponse struct {
    Body []byte 
    ContentType string 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/ec2/2016-11-15/typescript/src/sdk/models/operations/postterminateinstances.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostTerminateInstancesActionEnum {
    TerminateInstances = "TerminateInstances"
}

export enum PostTerminateInstancesVersionEnum {
    TwoThousandAndSixteen1115 = "2016-11-15"
}


export class PostTerminateInstancesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Action" })
  action: PostTerminateInstancesActionEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Version" })
  version: PostTerminateInstancesVersionEnum;
}


export class PostTerminateInstancesHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class PostTerminateInstancesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: PostTerminateInstancesQueryParams;

  @SpeakeasyMetadata()
  headers: PostTerminateInstancesHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=text/xml" })
  request?: Uint8Array;
}


export class PostTerminateInstancesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
